presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -756,6 +854,16 @@ public static boolean isTypeOnlyCoercion(Type actualType, Type expectedType)
             return ((VarcharType) actualType).getLength().orElse(Integer.MAX_VALUE) < ((VarcharType) expectedType).getLength().orElse(Integer.MAX_VALUE);
         }
 
+        if (actualType instanceof DecimalType && expectedType instanceof DecimalType) {
+            DecimalType actualDecimal = (DecimalType) actualType;
+            DecimalType expectedDecimal = (DecimalType) expectedType;
+
+            if (actualDecimal.isShort() ^ expectedDecimal.isShort()) {
+                return false;
+            }
+            return actualDecimal.getScale() == expectedDecimal.getScale() && actualDecimal.getPrecision() <= expectedDecimal.getPrecision();
```

Comment:
this probably should check if dest (digits before .) >= src (digits before .) and dest.scale >= src.scale